src/Codeception/Test/Metadata.php

```diff
@@ -19,71 +28,59 @@ class Metadata
         'incomplete' => null
     ];
 
+    /**
+     * @var array
+     */
     protected $current = [];
+    /**
+     * @var array
+     */
     protected $services = [];
-    protected $reports = [];
-
     /**
-     * @return mixed
+     * @var array
      */
+    protected $reports = [];
+
     public function getEnv()
     {
         return $this->params['env'];
     }
 
-    /**
-     * @return array
-     */
-    public function getGroups()
+    public function getGroups(): array
     {
         return array_unique($this->params['group']);
     }
 
-    /**
-     * @param mixed $groups
-     */
-    public function setGroups($groups)
```

Comment:
I don't know why the docblock said that $groups can be mixed.
The only 2 places that call it, pass array of strings.